src/components/forms/CloudInitExpandButton.tsx

```diff
@@ -0,0 +1,103 @@
+import type { FC } from "react";
+import { Button, Icon } from "@canonical/react-components";
+import { useModal } from "context/useModal";
+import ResourceLink from "components/ResourceLink";
+import type { CloudInitKey } from "components/forms/CloudInitForm";
+import CloudInitYamlEditorModal from "components/forms/CloudInitYamlEditorModal";
+import type { InstanceAndProfileFormikProps } from "components/forms/instanceAndProfileFormValues";
+import type { ConfigRowMetadata } from "util/configInheritance";
+import { ensureEditMode } from "util/instanceEdit";
+import classnames from "classnames";
+import { useIsScreenBelow } from "context/useIsScreenBelow";
+
+const LABELS: Record<CloudInitKey, string> = {
+  cloud_init_network_config: "cloud-init network config",
+  cloud_init_user_data: "cloud-init user data",
+  cloud_init_vendor_data: "cloud-init vendor data",
+};
+
+interface Props {
+  formik: InstanceAndProfileFormikProps;
+  project: string;
+  name: CloudInitKey;
+  metadata: ConfigRowMetadata;
+  initialValue: string;
+  onApplyChanges?: () => void;
+  isReadOnly?: boolean;
+  className?: string;
+}
+
+const CloudInitExpandButton: FC<Props> = ({
+  formik,
+  project,
+  name,
+  metadata,
```

Comment:
```suggestion
  source,
```
Shall we unpack the metadata object in the caller?

Reply:
Cool

<img width="794" height="340" alt="image" src="https://github.com/user-attachments/assets/3cfbc261-b18b-431c-bed1-f23d7b1821ad" />

However, when we click on the chip, it goes to the right page but underneath the modal. This is unexpected but what should be the correct behavior ?
Ctrl + click and mouse wheel click work and open a new tab.